crates/sail-common/src/spec/data_type.rs

```diff
@@ -67,23 +65,23 @@ impl DataType {
                 name: default_field_name.to_string(),
                 data_type: x,
                 nullable,
-                metadata: HashMap::new(),
+                metadata: vec![],
             }]),
         };
         Schema { fields }
     }
 }
 
-#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize, Hash)]
 #[serde(rename_all = "camelCase")]
 pub struct Field {
     pub name: String,
     pub data_type: DataType,
     pub nullable: bool,
-    pub metadata: HashMap<String, String>,
+    pub metadata: Vec<(String, String)>,
```

Comment:
Here are the reasons for this change:
1. Having `metadata` as `Vec<(String, String)>` ensures that the spec is deterministic.
2. I'd like to derive `Hash` for `DataType`, which is not possible if it contains `HashMap`.